src/node/interfaces.cpp

```diff
@@ -584,11 +585,11 @@ class NodeImpl : public Node
         if (context) {
             m_context_ref = *context;
         } else {
-            m_context_ref = std::nullopt;
+            m_context_ref = std::monostate{};
         }
     }
     NodeContext* m_context{nullptr};
-    CoreContext m_context_ref{std::nullopt};
+    CoreContext m_context_ref{std::monostate{}};
```

Comment:
d34ab741269e0042584584d022c4dce7a7f7a596: monostate makes it a default-constructed variant which holds a value of its first alternative, no need to initialize it explicitly anymore.
```suggestion
    CoreContext m_context_ref;
```